### ISSUE TITLE
Bot Buster

### ISSUE THREAD

**Issue by LavdrimIslami** [cis3296f24/applebaum-final-project-section-005-applebaum#26]
Keywords

Bot detection, social media, reddit, twitter, python/go, NLP, fraud detection, web scraping, API

## Project Abstract
I propose a software that will classify whether a selected user from a social media like Reddit or X(Twitter) is an automated account or a human. The program will scrape the users public profile, and based on their activity produce a risk score based on how confident the program is in its analysis. 

### High Level Requirement
The software will provide a point of contact for the user to assess if accounts are bots or not. The user will enter the url of a targets account, and from there the software will scrape the targets data, run it through an NLP model trained to detect anomalies synonymous with bot speech. The software will either be written in Python, data scraping libraries. The dataset used for detection will be either public lists of keywords or human reviewed ones. 

### Conceptual Design

The program will take in a given url to a targets account, and using libraries such as beautiful soup data would be stored as raw JSON on object storage (S3). To actually process the data and compare, a model made with a hugging face transformer will be used to generate a score. 

The software could either be a chrome extension (which would require Typescript), or can be a standalone application, maintaining the original python stack.

## Proof of Concept
http://github.com/LavdrimIslami/BotBuster

This proof of concept will consist of a program that gives a live demonstration allowing for a target to be identified as a bot or not. 

Compilation instructions to be added

-clone repository to local machine
-run pip install -r requirements.txt to ensure that all dependencies are installed
-run python file

## Background

In the age of AI generated content, the fear of the dead internet theory is now bigger than ever. This software will give users peace of mind knowing that the accounts they are interacting with over social media are real people as opposed to ai bots content farming for engagement and clicks. There already is a lot of work going into preventing bots in social media, such as cloudflare and their bot management(https://www.cloudflare.com/learning/bots/what-is-bot-management/), or how X was able to remove almost 2 million bots from their platform(https://x.com/nikitabier/status/1977446408136650785). Despite this, they are still prevalent. This software aims to make a better user experience, as we know that its impossible to remove all the bots, knowing who is who will maintain integrity. 

## Required Resources

Hardware:
-Laptop
- GPU capable of training a model, or a cloud computing platform 

Software
-Python 3
-any IDE
-BeautifulSoup library
-training data
- S3
- huggingface

**Comment by ElanReizas** [comment]
When I go through pages of reddit I think to myself theres too many bots on this platform, and we gotta get rid of them. I think this project has important real world use and that is why I want to work on this topic. I worked on research with AI which allows me to identify and help combat the bot flood.

**Comment by dagt03** [comment]
I think this project is really cool, definitely something that I wouldn't mind paying for. I feel like AI has greatly increased the amount of bots that show up on social media, it's starting to flood big platforms like Reddit and Twitter, and honestly it's not like they are providing good solutions to the problem. I like that it gives users a practical tool to check the authenticity of accounts with a confidence score (similar to what I'm thinking with my own proposal).

I can contribute by helping build and fine-tune the NLP model for better accuracy, optimizing the data scraping process. On the frontend, if this becomes a Chrome extension, I can help with the interface to make it easy for users to input URLs and view the results.

Something that can make the project even cooler would be a feature that flags accounts that are likely bots in real time while you are scrolling through your feed. Also it would be interesting to see a history of analyzed accounts.

**Comment by JerryLin-SWE** [comment]
I like how you’re focusing on giving users a quick way to check if an account is real or automated. It’s  super useful, and I think the idea of turning it into a Chrome extension would make it even more accessible for regular people instead of just developers.

I could help with testing the scraping and model results, maybe trying out different accounts and seeing how accurate the detection feels. I’m also interested in helping with the frontend side or UI if you decide to make a browser version. Very nice idea!
